OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -314,14 +328,31 @@ public HackyAI(HackyAIInfo info, ActorInitializer init)
 					&& !unit.Info.HasTraitInfo<HuskInfo>()
 					&& unit.Info.HasTraitInfo<ITargetableInfo>();
 
-			unitCannotBeOrdered = a => a.Owner != Player || a.IsDead || !a.IsInWorld;
+			unitCannotBeOrdered = UnitCannotBeOrdered;
 
 			foreach (var decision in info.PowerDecisions)
 				powerDecisions.Add(decision.OrderName, decision);
 
 			maximumCaptureTargetOptions = Math.Max(1, Info.MaximumCaptureTargetOptions);
 		}
 
+		bool UnitCannotBeOrdered(Actor a)
+		{
+			if (a.Owner != Player || a.IsDead || !a.IsInWorld)
+				return true;
+
+			var tag = a.TraitOrDefault<ScriptTags>();
```

Comment:
That's currently used by APC rush. They shouldn't be in control of default HackyAI's otherwise they will keep returning to base. Same goes for any future teams with complex tasks. Tagging looked good, as it doesn't require introduction of new Lua APIs, just for HackyAI implementation of AI.